File: coursework/infrastructure/rest-api/src/main/java/com/konstantion/controllers/product/dto/ProductDto.java
package com.konstantion.controllers.product.dto;

import java.time.LocalDateTime;
import java.util.UUID;

public record ProductDto(
        UUID id,
        String name,
        Double price,
        Double weight,
        UUID categoryId,
        String description,
        UUID creatorId,
        byte[] imageBytes,
        LocalDateTime createdAt,
        LocalDateTime deactivateAt,
        Boolean active
) {
}

File: coursework/frontend/src/app/models/dto/product/product-dto.ts
export interface ProductDto {
    id: string;
    name: string;
    price: number;
    weight: number;
    categoryId: string;
    description: string;
    creatorId: string;
    imageBytes: ArrayBuffer;
    createdAt: Date;
    deactivateAt: Date;
    active: boolean;
}